src/huggingface_hub/utils/_hf_folder.py

```diff
@@ -14,22 +14,31 @@
 # limitations under the License.
 """Contain helper class to retrieve/store token from/to local cache."""
 import os
+import warnings
 from pathlib import Path
 from typing import Optional
 
+from .. import constants
+
 
 class HfFolder:
-    path_token = Path("~/.huggingface/token").expanduser()
+    path_token = Path(constants.HF_TOKEN_PATH)
+    # Private attribute. Will be removed in v0.15
```

Comment:
I don't think you will be able to ever get rid of this ;-) You should support migrating from old tokens to new tokens for a long time IMO